numpy/core/fromnumeric.py

```diff
@@ -869,6 +869,14 @@ def sort(a, axis=-1, kind=None, order=None):
         but unspecified fields will still be used, in the order in which
         they come up in the dtype, to break ties.
 
+    by: tuple or object, optional
+        With a given array `a` sort the elements not by arithmetic
+        order, but by any arbitrary order defined by tuple, there can
+        be multiple keys in a tuple form , or just a single object.
+        Tuple contains k (N,)-shaped sequences
+        The `k` different "columns" to be sorted.  The last column (or row if
+        `keys` is a 2D array) is the primary sort key.
```

Comment:
@hameerabbasi, this is copied from lexsort, but I am not sure I agree with it.  In general, I think I would think of it by internally reversing the sort priority (so that the first one is the most prominent one).  To be honest, the only counter argument to that is that lexsort is being weird about it...  I am almost curious whether we should allow `np.argsort(a, by=...)` (although `a` is not really needed for indirect lexsorting...).